### PR TITLE
B feature/add leave profile type

### PR DESCRIPTION
**Pull request by mariobasic**

### CODE REVIEW

**Comment by stanislavgrujic** [comment]
@JonathanTolic Please review.

File: src/main/java/eu/execom/hawaii/jobs/IncrementServiceYearsJob.java [hunk 1]
```diff
@@ -45,18 +49,18 @@ public void addServiceYearsToUser() {
     Supplier<Stream<User>> userStream = () -> users.stream()
                                                    .filter(user -> startedWorkingToday(
                                                        MonthDay.from(user.getStartedWorkingDate())));
+    userStream.get().forEach(user -> {
+      user.setYearsOfService(user.getYearsOfService() + 1);
+      userRepository.save(user);
```
Comment (stanislavgrujic) [inline src/main/java/eu/execom/hawaii/jobs/IncrementServiceYearsJob.java]: You also have the `save()` method in the `userService`, there is no need to introduce an additional dependency in this class. Please try to know the classes which you use as much as possible.

File: src/main/java/eu/execom/hawaii/model/enumerations/LeaveProfileType.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package eu.execom.hawaii.model.enumerations;
+
+public enum LeaveProfileType {
+  ZERO_TO_FIVE_YEARS, FIVE_TO_TEN_YEARS, TEN_TO_FIFTEEN_YEARS, FIFTEEN_TO_TWENTY_YEARS,
+  TWENTY_TO_TWENTYFIVE_YEARS, TWENTYFIVE_TO_THIRTY_YEARS, THIRTY_TO_THIRTYFIVE_YEARS,
+  THIRTYFIVE_TO_FORTY_YEARS, PART_TIME, DEFAULT
```
Comment (stanislavgrujic) [inline src/main/java/eu/execom/hawaii/model/enumerations/LeaveProfileType.java]: Break enum values on in each line, it is easier to read it.

**Comment by mariobasic** [comment]
Should I change anything else? I want to make new pull request but I need changes made here.

File: src/main/java/eu/execom/hawaii/jobs/IncrementServiceYearsJob.java [hunk 3]
```diff
@@ -69,8 +70,39 @@ private boolean startedWorkingToday(MonthDay startedWorkingDate) {
   }
 
   private Predicate<User> shouldUpdateLeaveProfile() {
-    return user -> user.getYearsOfService() == FIVE_YEARS ||
-                   user.getYearsOfService() == TEN_YEARS  ||
-                   user.getYearsOfService() == FIFTEEN_YEARS;
+    return user -> user.getYearsOfService() % UPDATE_THRESHOLD == 0;
+  }
+
+  private LeaveProfile nextLeaveProfile(LeaveProfile currentLeaveProfile) {
```
Comment (stanislavgrujic) [inline src/main/java/eu/execom/hawaii/jobs/IncrementServiceYearsJob.java]: Please rename to `findNextLeaveProfile`. Reasoning: method names should have verb imperative in naming, in most cases.

**Pull request merged**